### PR TITLE
Add new package: dovecot

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

**Review by michaelkuhn — CHANGES_REQUESTED** [review]
If we use the proper release tarballs, we can get rid of the autotools dependencies.

Also, according to https://wiki2.dovecot.org/CompilingSource, Dovecot probably needs at least OpenSSL and has some more optional dependencies.

**Comment by adamjstewart** [comment]
Can you add the missing dependencies?

**Comment by darmac** [comment]
> Can you add the missing dependencies?

I added those dependencies:
```
    depends_on('pkg-config', type='build')
    depends_on('gettext',    type='build')
    depends_on('pandoc',     type='build')
```

But `pandoc` build fail because `texlive` build failed.
```
==> [2020-10-19-14:23:50.999527] '/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/texlive-20200406-gzo55ewgkzkkaeowcehiyep4kixfjbfv/bin/aarch64-linux/mtxrun' '--generate'
/usr/bin/env: 'texlua': No such file or directory
==> Error: ProcessError: Command exited with status 127:
    '/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/texlive-20200406-gzo55ewgkzkkaeowcehiyep4kixfjbfv/bin/aarch64-linux/mtxrun' '--generate'

1 error found in build log:
     41583    fmtutil [INFO]: /home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.1/texlive-20200406
              -gzo55ewgkzkkaeowcehiyep4kixfjbfv/texmf-var/web2c/tex/lollipop.fmt installed.
     41584    fmtutil [INFO]: disabled formats: 5
     41585    fmtutil [INFO]: successfully rebuilt formats: 55
     41586    fmtutil [INFO]: total formats: 60
     41587    fmtutil [INFO]: exiting with status 0
     41588    ==> [2020-10-19-14:23:50.999527] '/home/xiaojun/spack/opt/spack/linux-centos8-aarch64/gcc-8.2.
              1/texlive-20200406-gzo55ewgkzkkaeowcehiyep4kixfjbfv/bin/aarch64-linux/mtxrun' '--generate'
  >> 41589    /usr/bin/env: 'texlua': No such file or directory

See build log for details:
  /tmp/root/spack-stage/spack-stage-texlive-20200406-gzo55ewgkzkkaeowcehiyep4kixfjbfv/spack-build-out.txt

==> Warning: Skipping build of pandoc since texlive failed
==> Warning: Skipping build of dovecot since pandoc failed
==> Error: Installation of dovecot failed.  Review log for details
```

**Comment by adamjstewart** [comment]
In that case, you should still add the dependencies so this can be merged, but you can open a separate issue to try to figure out how to solve the texlive build issue.

**Comment by darmac** [comment]
I opened an issue for `texlive`, https://github.com/spack/spack/issues/19704
I'll keep testing this patch after this issue be fixed.

**Comment by michaelkuhn** [comment]
How about adding a `doc` variant (that defaults to `False`) so we can get rid of the `pandoc`/`texlive` dependencies by default? I guess `pandoc` is only required for building the documentation?

**Comment by darmac** [comment]
Updated in https://github.com/spack/spack/pull/18775/commits/3257ca591e9c4dba3b0691cb12a2511499479290

File: var/spack/repos/builtin/packages/dovecot/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class Dovecot(AutotoolsPackage):
+    """Dovecot mail server."""
+
+    homepage = "https://www.dovecot.org/"
+    url      = "https://dovecot.org/releases/2.3/dovecot-2.3.11.3.tar.gz"
+
+    version('2.3.11.3', sha256='d3d9ea9010277f57eb5b9f4166a5d2ba539b172bd6d5a2b2529a6db524baafdc')
+
+    depends_on('pkg-config', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/dovecot/package.py]:
```suggestion
    depends_on('pkgconfig', type='build')
```
We have a virtual package for this

**Review by adamjstewart — APPROVED** [review]

**Comment by michaelkuhn** [comment]
I still think we should add an OpenSSL dependency. Otherwise, I am not sure how useful this is. Would you use a mail server without encryption? From a quick test, adding `depends_on('openssl')` seems to be enough.

**Comment by alalazo** [comment]
@darmac Can you address https://github.com/spack/spack/pull/18775#issuecomment-753861905 ?

**Comment by alalazo** [comment]
Closing as stale

**Pull request closed without merging**